src/sentry/api/validators/notifications.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Dict, Mapping, Optional, Set, Union
+
+from sentry.api.exceptions import ParameterValidationError
+from sentry.models.integration import ExternalProviders
+from sentry.notifications.helpers import validate as helper_validate
+from sentry.notifications.types import (
+    NotificationScopeType,
+    NotificationSettingOptionValues,
+    NotificationSettingTypes,
+)
+
+
+def intersect_dict_set(d: Dict[int, Any], s: Set[int]) -> Dict[int, Any]:
+    """ Get the sub-dictionary where the keys are in the intersections the original keys and the set. """
+    # TODO(mgaeta): Genericize the key type and move to sentry.utils.
+    return {k: v for k, v in d.items() if k in s}
+
+
+def validate_organizations(
+    organization_ids_to_look_up: Set[int],
+    user: Optional[Any] = None,
+    team: Optional[Any] = None,
+) -> Mapping[int, Any]:
+    if not organization_ids_to_look_up:
+        return {}
+
+    if user:
+        organizations_by_id = {organization.id: organization for organization in user.get_orgs()}
+    elif team:
+        organizations_by_id = {team.organization.id: team.organization}
+    else:
+        raise Exception("User or Team must be set.")
+
+    missing_organization_ids = organization_ids_to_look_up - organizations_by_id.keys()
```

Comment:
when does this happen? or is it just to be safe?

Reply:
This is just to be safe. I don't think anything bad would happen if a user had a notification setting for a project or organization they were not a member of.